src/Dependencies/PooledObjects/PooledStringBuilder.cs

```diff
@@ -72,10 +72,18 @@ public string ToStringAndFree(int startIndex, int length)
         private static readonly ObjectPool<PooledStringBuilder> s_poolInstance = CreatePool();
 
         // if someone needs to create a private pool;
-        public static ObjectPool<PooledStringBuilder> CreatePool()
+        /// <summary>
+        /// If someone need to create a private pool
+        /// </summary>
+        /// <param name="size">Initial size of the pool. Bounded to 32 &lt;= Size &lt;= 256)</param>
+        /// <returns></returns>
+        public static ObjectPool<PooledStringBuilder> CreatePool(int size = 32)
         {
+            const int LowerBound = 32;
+            const int UpperBound = 256;
             ObjectPool<PooledStringBuilder> pool = null;
-            pool = new ObjectPool<PooledStringBuilder>(() => new PooledStringBuilder(pool), 32);
+            size = (size < LowerBound ? LowerBound: (size > UpperBound ? UpperBound : size));
```

Comment:
Please adjust the comment accordingly as well.

---
In reply to: [103270501](https://github.com/dotnet/roslyn/pull/17134#discussion_r103270501) [](ancestors = 103270501)